ludwig/models/trainer.py

```diff
@@ -99,6 +100,7 @@ def __init__(
         model: ECD,
         optimizer=None,
         epochs=100,
+        save_every_n_steps=0,
```

Comment:
Nit, but the name here seems off to me. I would probably prefer something like `steps_per_checkpoint` or `steps_per_eval`.

Reply:
In the current implementation, it looks like checkpointing, saving, and eval are all happening at the same time based on this param: https://github.com/ludwig-ai/ludwig/pull/1803/files#diff-5f0ad4ede44009f6f1337ae0ce8dc2440d29750e58a3f6a171167d35474bfd1dR1030

One problem with the current nomenclature, as things are implemented, is that if the user sets `skip_model_save=True`, then `steps_per_save` still has an effect (controlling checkpoint frequency), which seems counterintuitive.

My understanding is that the desired state should be that every time we hit `steps_per_save` we checkpoint the model and kickoff an eval request to evaluate on that checkpoint. The eval may happen some time later, but it will still happen on that checkpoint.

Not sure why we have model saving and checkpoint separate to be honest. I don't think model saving is actually necessary until the very end of training, assuming we want to save the model in a "frozen" state (without optimizer, etc.). I would instead think of the overall process as: checkpoint the model every n steps, kickoff eval on the checkpoint, and at the end of training select the best model and save it to the top-level model directory.

If that makes sense to you as well, then I would say probably `steps_per_checkpoint` feels the most appropriate to me.

Reply:
Thinking about this more, `steps_per_checkpoint` seems good to me, since asynchronous evaluation occurs certainly on a specific-checkpoint basis, and model saving and checkpointing are practically synonymous, acknowledging that we do an extra "frozen" save (after evaluation) whenever we encounter a model with a higher validation score.